shared-module/bitbangio/SPI.c

```diff
@@ -36,6 +36,52 @@
 
 #define MAX_BAUDRATE (common_hal_mcu_get_clock_frequency() / 48)
 
+STATIC uint8_t bitrev(uint8_t n) {
+    uint8_t r = 0;
+    for(int i=0;i<8;i++) r |= ((n>>i) & 1)<<(7-i);
+    return r;
+}
```

Comment:
FYI (nothing to change): clang has a builtin for this, because it's a single instruction in some architectures.
https://gcc.gnu.org/bugzilla/show_bug.cgi?id=50481

Reply:
CMSIS provides __RBIT for this purpose, across compilers but specific to ARM.  Hypothetically, conditionally include "cmsis_compiler.h", then
```
STATIC uint8_t bitrev(uint8_t n) {
#ifdef __arm__
    return __RBIT(n) >> 24;
#else
    uint8_t r = 0;
    for(int i=0;i<8;i++) r |= ((n>>i) & 1)<<(7-i);
    return r;
#endif
}
```
saves 48 bytes.